src/mod/databases/mod_mariadb/mod_mariadb.c

```diff
@@ -627,13 +627,25 @@ switch_status_t mariadb_send_query(mariadb_handle_t *handle, const char* sql)
 {
 	char *err_str;
 	int ret;
+	unsigned retries = 60; //60 tries, will take 30 to 60 seconds at worst
 
 	switch_safe_free(handle->sql);
 	handle->sql = strdup(sql);
+    again:
 	handle->stored_results = 0;
 	ret = mysql_real_query(&handle->con, sql, (unsigned long)strlen(sql));	
 	if (ret) {
 		err_str = mariadb_handle_get_error(handle);
+		if (strstr(err_str, "Deadlock found when trying to get lock; try restarting transaction")) {
+			if (--retries > 0) {
+				switch_safe_free(err_str);
+				// We are waiting for 500 ms and random time is not more than 500 ms. 
+				// This is necessary so that the delay on the primary and secondary servers does not coincide and deadlock does not occur again.
+				switch_yield(500+rand()%500);
```

Comment:
Just a small suggestion: if it is possible to avoid using modulo operation then we should. how about using `and` with a power of 2 (where 512 is 2^9) instead?
`switch_yield(500 + (rand() & 511))`

Also we should use `/* */` comment style.
Also we should make a new line after an `if` block.

Reply:
Fixing